### PR TITLE
Fork devcert

### PR DESCRIPTION
**Pull request by xerxovksi**
## Description

Fork [devcert](https://github.com/davewasmer/devcert.git) and add support to revoke an SSL certificate created by **devcert**.

## Motivation and Context

When a developer inputs a hostname for which there is an SSL configured automatically, ccweb-add-on-ssl should remove the old SSL certificate and create a new one, after the developer accepts through the prompt to do so.

However, the npm package - `devcert` which we are are using to manage automatic SSL in the Dev Tools CLI, does not support revoking the older SSL certificates.

To support this, we have created this fork of `devcert` and addd this functionality.

## Change Details
In `src/certificates.ts` file:
exported a new function called `revokeDomainCertificate()` to revoke the certificates passed in domains[] parameter of this function, by calling:

```
openssl(['ca', '-config', domainCertConfigPath, '-revoke', domainCertPath, '-keyfile', caKeyPath, '-cert', caCertPath]) 
```

before deleting the certificate from the directory.

## How Has This Been Tested?

- Package with this change was published to an internal registry.
- Installed this package in an Adobe Express add-on to create the SSL certificate.
- Through the same add-on, re-created an SSL certificate for the same domain by revoking the older one first.
- Hosted the add-on locally using this SSL certificate.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: LICENSE
Comment (kerrishotts) [inline LICENSE:1]: We can't take over the copyright here; so while a portion of our changes are (C) Adobe, the original copyright holder needs to be listed here too.

So I'd do something like this:

```
(C) Copyright Dave Wasmer
Portions (C) Copyright 2023 Adobe.
```

File: README copy.md
Comment (kerrishotts) [inline README copy.md:1]: In our fork notice, let's include the orignal author's name & copoyright


(i) Note

This is a fork of devcert (Copyright Dave Wasmer). (etc.)

**Review by kerrishotts — CHANGES_REQUESTED** [review]

File: LICENSE
Comment (xerxovksi) [reply LICENSE:1]: Updated.

File: README copy.md
Comment (xerxovksi) [reply README copy.md:1]: Added.

**Review by kerrishotts — APPROVED** [review]
LGTM

**Pull request merged**